src/index.js

```diff
@@ -4,556 +4,608 @@ const codec = require('./codec')
+    const names = this.protoNames()
+    const parts = this.toString().split('/').slice(1)
+
+    if (parts.length < 4) {
+      throw new Error('multiaddr must have a valid format: "/{ip4, ip6, dns4, dns6}/{address}/{tcp, udp}/{port}".')
+    } else if (codes[0] !== 4 && codes[0] !== 41 && codes[0] !== 54 && codes[0] !== 55) {
+      throw new Error(`no protocol with name: "'${names[0]}'". Must have a valid family name: "{ip4, ip6, dns4, dns6}".`)
+    } else if (parts[2] !== 'tcp' && parts[2] !== 'udp') {
+      throw new Error(`no protocol with name: "'${names[1]}'". Must have a valid transport protocol: "{tcp, udp}".`)
+    }
 
-/**
- * Gets a Multiaddrs node-friendly address object. Note that protocol information
- * is left out: in Node (and most network systems) the protocol is unknowable
- * given only the address.
- *
- * Has to be a ThinWaist Address, otherwise throws error
- *
- * @returns {{family: string, address: string, port: number}}
- * @throws {Error} Throws error if Multiaddr is not a Thin Waist address
- * @example
- * Multiaddr('/ip4/127.0.0.1/tcp/4001').nodeAddress()
- * // {family: 'IPv4', address: '127.0.0.1', port: '4001'}
- */
-Multiaddr.prototype.nodeAddress = function nodeAddress () {
-  const codes = this.protoCodes()
-  const names = this.protoNames()
-  const parts = this.toString().split('/').slice(1)
-
-  if (parts.length < 4) {
-    throw new Error('multiaddr must have a valid format: "/{ip4, ip6, dns4, dns6}/{address}/{tcp, udp}/{port}".')
-  } else if (codes[0] !== 4 && codes[0] !== 41 && codes[0] !== 54 && codes[0] !== 55) {
-    throw new Error(`no protocol with name: "'${names[0]}'". Must have a valid family name: "{ip4, ip6, dns4, dns6}".`)
-  } else if (parts[2] !== 'tcp' && parts[2] !== 'udp') {
-    throw new Error(`no protocol with name: "'${names[1]}'". Must have a valid transport protocol: "{tcp, udp}".`)
+    return {
+      family: (codes[0] === 41 || codes[0] === 55) ? 6 : 4,
+      address: parts[1],
+      port: parseInt(parts[3]) // tcp or udp port
+    }
   }
 
-  return {
-    family: (codes[0] === 41 || codes[0] === 55) ? 6 : 4,
-    address: parts[1], // ip addr
-    port: parseInt(parts[3]) // tcp or udp port
+  // TODO find a better example, not sure about it's good enough
+  /**
+   * Returns if a Multiaddr is a Thin Waist address or not.
+   *
+   * Thin Waist is if a Multiaddr adheres to the standard combination of:
+   *
+   * `{IPv4, IPv6}/{TCP, UDP}`
+   *
+   * @param {Multiaddr} [addr] - Defaults to using `this` instance
+   * @returns {boolean} isThinWaistAddress
+   * @example
+   * ```js
+   * const mh1 = Multiaddr('/ip4/127.0.0.1/tcp/4001')
+   * // <Multiaddr 047f000001060fa1 - /ip4/127.0.0.1/tcp/4001>
+   * const mh2 = Multiaddr('/ip4/192.168.2.1/tcp/5001')
+   * // <Multiaddr 04c0a80201061389 - /ip4/192.168.2.1/tcp/5001>
+   * const mh3 = mh1.encapsulate(mh2)
+   * // <Multiaddr 047f000001060fa104c0a80201061389 - /ip4/127.0.0.1/tcp/4001/ip4/192.168.2.1/tcp/5001>
```

Comment:
Per the TODO above: you can for example add a dnsaddr multiaddr or a webrtc-star multiaddr here

Reply:
Can you add here for example: `/ip4/127.0.0.1/tcp/2000/wss/p2p-webrtc-star/p2p/QmcgpsyWgH8Y8ajJz1Cu72KnS5uo2Aa2LpzU7kinSooo2a` 
You can also remove the TODO above

Reply:
can i please suggest a change so i dont make mistakes ?